toolboxes/feel/feelmodels/modelvf/fluidmecconvectiveterm.hpp

```diff
@@ -0,0 +1,843 @@
+
+    size_type dynamicContext() const { return 0; }
+
+    uint16_type polynomialOrder() const
+        {
+            if ( this->applyLinearization() )
+                return expr_evaluate_velocity_linearization_opertors_type::polynomialOrderId()+expr_evaluate_velocity_opertors_type::polynomialOrderGrad();
+            else
+                return expr_evaluate_velocity_opertors_type::polynomialOrderId()+expr_evaluate_velocity_opertors_type::polynomialOrderGrad();
+        }
+
+    template<typename Geo_t, typename Basis_i_t, typename Basis_j_t>
+    using tensor_base_type = tensorBase<Geo_t,Basis_i_t,Basis_j_t,shape_type,value_type>;
+
+    template<typename Geo_t, typename Basis_i_t, typename Basis_j_t>
+    struct tensor : public this_type::template tensor_base_type<Geo_t,Basis_i_t,Basis_j_t>
+    {
+    protected :
+        using super_type = typename this_type::template tensor_base_type<Geo_t,Basis_i_t,Basis_j_t>;
+    public :
+
+        typedef typename super_type::gmc_type gmc_type;
+        typedef typename super_type::gm_type gm_type;
+        typedef typename super_type::value_type value_type;
+
+        using shape = typename super_type::shape_type;
+        using matrix_shape_type = typename super_type::matrix_shape_type;
+
+        using array_shape_type = typename super_type::new_array_shape_type;
+        using ret_type = typename super_type::ret_type;
+
+        template<typename... TheArgsType>
+        tensor( this_type const& expr, Geo_t const& geom, const TheArgsType&... theInitArgs )
+            :
+            super_type( geom, theInitArgs... )
+            {}
+
+        void update( Geo_t const& geom ) override { CHECK( false ) << "should be override"; };
+        void update( Geo_t const& geom, Basis_i_t const& fev, Basis_j_t const& feu ) override { CHECK( false ) << "should be override"; }
+        void update( Geo_t const& geom, Basis_i_t const& fev ) override { CHECK( false ) << "should be override"; }
+
+        template<typename TheExprExpandedType,typename TupleTensorSymbolsExprType, typename... TheArgsType>
+        void update( std::true_type /**/, TheExprExpandedType const& exprExpanded, TupleTensorSymbolsExprType & ttse,
+                     Geo_t const& geom, const TheArgsType&... theUpdateArgs );
+
+
+        value_type
+        evaliq( uint16_type i, uint16_type c1, uint16_type c2, uint16_type q ) const override
+            {
+                if constexpr ( this_type::is_applied_as_eval )
+                    return this->evalq( c1,c2,q );
+                else
+                {
+                    CHECK( false) << "not allow";
+                    return value_type(0);
+                }
+            }
+        ret_type
+        evaliq( uint16_type i, uint16_type q ) const override
+            {
+                if constexpr ( this_type::is_applied_as_eval )
+                    return this->evalq( q );
+                else
+                {
+                    CHECK( false) << "not allow";
```

Comment:
throw an exception here

Reply:
or better message ?

Reply:
this is in an expression tensor, so no exception here